homeassistant/components/sensor/__init__.py

```diff
@@ -246,6 +252,20 @@ def device_class(self) -> SensorDeviceClass | None:
             return self.entity_description.device_class
         return None
 
+    @final
+    @property
+    def is_numeric(self) -> bool:
+        """Return true if the sensor is numeric."""
+        if (
+            self.state_class is not None
+            or self.unit_of_measurement is not None
```

Comment:
I think this should use the native unit.
```suggestion
            or self.native_unit_of_measurement is not None
```